code-challenges.test.js

```diff
@@ -27,41 +27,88 @@ const numbersArray2 = [24, 27, 30, 33, 36]
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+// describe("capNouns", () => {
+//     it("takes in an array and returns the same array capitalised", () => {
+//         const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+//         const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+//         expect(capNouns(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+//         expect(capNouns(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+//     })
+// })
+
+// Good fauilure:  ReferenceError: capNouns is not defined
+
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+// Pseudo code: write a function that takes in the array. use .map to iterate through the array, create an action, below return the value 0 indexed, this will access the first index of each value in the array. Use .toUpperCase to captialise the first index letter in the array, then concat the rest of the remaining word them with + (also .substring(1)) .
 
+// const capNouns = (array) => {
+//     return array.map((value) => {
+//         return value[0].toUpperCase() + value.substring(1)
+//     })
+// }
+// console.log(upCase(randomNouns1))
+// console.log(upCase(randomNouns2))
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
-
-// b) Create the function that makes the test pass.
-
-// Pseudo code:
+describe("firstVowel", () => {
+    it("finds the first vowel in a strings returns it's index", () => {
+        const vowelTester1 = "learn"
+        const vowelTester2 = "academy"
+        const vowelTester3 = "challenges"
+        expect(firstVowel(vowelTester1)).toEqual(1)
+        expect(firstVowel(vowelTester2)).toEqual(0)
+        expect(firstVowel(vowelTester3)).toEqual(2)
+    })
+})
+// // Expected output: 1
+// // Expected output: 0
+// // Expected output: 2
+
+// // b) Create the function that makes the test pass.
+
+// Pseudo code: create a for function/loop to iterate through the string. set the condition for the loop to stop and return the index on the first vowl it comes accross. (this one was a struggle and I couldn't get it right, I tried for 6+ hours.
```

Comment:
Totally fine to struggle with a question. Not sure what your approach was though with the given information on the assessment, so I can't give you any pointers. Make sure to pay attention during review and ask questions if you have them!